packages/collector/test/tracing/database/elasticsearch_modern/test.js

```diff
@@ -31,58 +32,54 @@ mochaSuiteFn('tracing/elasticsearch (modern client)', function () {
   this.timeout(Math.max(config.getTestTimeout() * 4, 30000));
 
   before(() => {
-    originalEsVersion = require('@elastic/elasticsearch/package.json').version;
+    // NOTE: We cannot use `require(@elastic/elasticsearch/package.json), because of
+    //       https://github.com/elastic/elasticsearch-js/blob/v7.16.0/package.json#L11
+    originalEsVersion = JSON.parse(
+      fs.readFileSync(`${path.dirname(require.resolve('@elastic/elasticsearch'))}/package.json`)
+    ).version;
+
     // eslint-disable-next-line no-console
     console.log(`original version: @elastic/elasticsearch@${originalEsVersion}`);
   });
 
   after(done => installLibraryVersion(originalEsVersion, done));
 
+  /**
+   * transport: instrumentation > 7.9.1
+   * api: instrumentation < 7.9.1
+   *
+   * The words "transport" and "api" try to describe the different
+   * mechanismn we use in core/src/tracing/instrumentation/database/elasticsearchModern.js
+   */
   [
-    //
     {
-      versionRange: '^7.9.1',
-      apiVersion: '7.9',
+      versionRange: 'latest',
       instrumentationFlavor: 'transport'
     },
     {
       versionRange: '7.9.0',
-      apiVersion: '7.9',
       instrumentationFlavor: 'api'
     }
-
-    // To test older client versions, you also need to set up a matching version of the Elasticsearch Docker container.
-    // The legacy client (package `elasticsearch`) supported using different server/API versions with the same client
-    // library, the modern client (`@elastic/elasticsearch`) does not.
-    // See docker-compose.yml for how to start older ES servers.
-    //
-    // Tests for older versions can be run manually, but are not executed on CI.
-    //
-    // {
-    //   versionRange: '^5.6.22',
-    //   apiVersion: '5.6',
-    //   instrumentationFlavor: 'api'
-    // }
-    //
-    // {
-    //   versionRange: '^6.8.8',
-    //   apiVersion: '6.8',
-    //   instrumentationFlavor: 'api'
-    // }
-  ].forEach(({ versionRange: esClientVersionRangeUnderTest, apiVersion, instrumentationFlavor }) => {
+  ].forEach(({ versionRange: esClientVersionRangeUnderTest, instrumentationFlavor }) => {
     describe(
       // eslint-disable-next-line no-useless-concat
       `@elastic/elasticsearch@${esClientVersionRangeUnderTest}/` + `instrumentation flavor: ${instrumentationFlavor}`,
       function () {
-        before(done => installLibraryVersion(esClientVersionRangeUnderTest, done));
+        before(done => {
+          if (esClientVersionRangeUnderTest === 'latest') {
+            return done();
+          }
+
+          installLibraryVersion(esClientVersionRangeUnderTest, done);
+        });
 
         globalAgent.setUpCleanUpHooks();
         const agentControls = globalAgent.instance;
 
         const controls = new ProcessControls({
           dirname: __dirname,
           useGlobalAgent: true,
-          env: { ES_API_VERSION: apiVersion }
+          env: {}
```

Comment:
_nitpick_: If we do not pass any environment variables, we could also remove this line completely, I think (instead of passing an empty object).